core/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -86,7 +83,7 @@ class StorageLevel private(
       false
   }
 
-  def isValid: Boolean = (useMemory || useDisk || useOffHeap) && (replication > 0)
+  def isValid: Boolean = (useMemory || useDisk) && (replication > 0)
```

Comment:
No, because

1. `offHeap` does not necessarily imply `useDisk`, and
2. We already perform well-formedness checks at the start of the constructor (e.g. offHeap == true implies deserialized == false).